src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1187,6 +1187,46 @@ private static DateTime ConvertToUTC(DateTime dateTime, TimeZoneInfo fromTimeZon
             return resultDateTime.UtcDateTime;
         }
 
+        public static FormulaValue Unichar(IRContext irContext, NumberValue[] args)
+        {
+            double number = args[0].Value;
+
+            // Unicode upper and lower bounds
+            const double lowerBound = 1; // Excel returns error for Unichar(0)
+            const double upperBound = 1114111; // 0x10FFFF
+
+            // https://unicode.org/glossary/#surrogate_pair
+            const double surrogateMin = 55296; // 0xD800
+            const double surrogateMax = 57343; // 0xDFFF
+
+            bool isOutOfRange = number < lowerBound || number > upperBound;
+            bool isPartialSurrogate = number >= surrogateMin && number <= surrogateMax;
+
+            if (isOutOfRange)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = $"Input value {number} falls outside the allowable range",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.InvalidArgument
+                });
+            }
+            else if (isPartialSurrogate)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = $"Input value {number} is an invalid argument",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.InvalidArgument
+                });
+            }
+            else
+            {
+                string unicode = char.ConvertFromUtf32((int)Math.Floor(number));

```

Comment:
Numeric arguments are rounded down for this function, so `UniChar(1114111)` should be the same as `UniChar(1114111.99)` - but this is not the case with the current implementation (it would return an error)

Reply:
I thought `Math.Floor` would help there and tried this test https://github.com/microsoft/Power-Fx/pull/2133/files#diff-1ac69fe955d21389cbada90d26edbc3abd7a4692f8f6c9a975e8fef93b068c8aR68-R69 , https://github.com/microsoft/Power-Fx/pull/2133/files#diff-1ac69fe955d21389cbada90d26edbc3abd7a4692f8f6c9a975e8fef93b068c8aR13-R14 . But i see what you mean. I will add this specific recommended test case and fix